### PR TITLE
Fixing Jira Issue : APIMANAGER-3556

### PR DESCRIPTION
**Pull request by chamindias**
Displaying a warning message when the visibility of the API is changed.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag [hunk 1]
```diff
@@ -96,6 +96,24 @@
     </div>
   <%  }%>
 
+<script>
+    $(document).ready(function(){
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: space needed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag [hunk 2]
```diff
@@ -96,6 +96,24 @@
...
+        <% if(api != null && api.subs == 0){ %>
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: space needed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag [hunk 3]
```diff
@@ -96,6 +96,24 @@
...
+            $('#visibility').change(function(){
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: space needed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag [hunk 4]
```diff
@@ -96,6 +96,24 @@
...
+                var visibility = $('#visibility').find(":selected").val();
+                if (visibility != "public"){
+                    jagg.message({content:"You have changed visibility of API, so based on visibility subscription availability may change",type:"warn"});
+                }
+                if (visibility == "public" || visibility == "private" || visibility == "controlled"){
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: space needed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag (hunk 1 above)
Comment (chamindias) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: Done in PR : https://github.com/wso2/carbon-apimgt/pull/1335

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag (hunk 2 above)
Comment (chamindias) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: Done in PR : https://github.com/wso2/carbon-apimgt/pull/1335

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag (hunk 3 above)
Comment (chamindias) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: Done in PR : https://github.com/wso2/carbon-apimgt/pull/1335

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag (hunk 4 above)
Comment (chamindias) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/default/templates/item-design/template.jag]: Done in PR : https://github.com/wso2/carbon-apimgt/pull/1335

**Pull request merged**